ci/code_checks.sh

```diff
@@ -179,6 +179,10 @@ if [[ -z "$CHECK" || "$CHECK" == "patterns" ]]; then
     invgrep -R --include="*.py" -E "super\(\w*, (self|cls)\)" pandas
     RET=$(($RET + $?)) ; echo $MSG "DONE"
 
+    MSG='Check for use of builtin filter function' ; echo $MSG
+    invgrep -R --include="*.py" -P '(?<!def)[\(\s]filter\(' pandas
```

Comment:
```suggestion
    invgrep -R --include="*.py" -P '(?<!def)filter\(' pandas
```

Is this necessary?

Reply:
that would fail on uses of `.filter`, `simplefilter`, `.apply_filter` and more